### DISCUSSION TITLE
Can I use aim as a lib in my program?

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [mihaigalos/aim#146] [Q&A] [answered]
I just came across aim today and it's very useful. I want to know can I embed its download feature in my program? Can it be used as a lib? Thanks.

**Comment by mihaigalos** [comment, marked as answer]
Hi,

`aim` is written as an executable wrapper around a [lib](https://github.com/mihaigalos/aim/blob/main/src/lib.rs).
The answer is yes.

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
Thank you for the clarification!

**Reply by mihaigalos** [reply]
Please try it out and let's continue this discussion if you have any problems.

**Reply by mihaigalos** [reply]
Hi @0x7FFFFFFFFFFFFFFF, is this conversation done? Can we please resolve it or do you need anything else?

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
Yes it's done. Thanks again my friend!
